ops/testing.py

```diff
@@ -342,14 +386,24 @@ def __init__(self, unit_name):
         self._unit_status = None
 
     def relation_ids(self, relation_name):
-        return self._relation_ids_map[relation_name]
+        try:
+            return self._relation_ids_map[relation_name]
+        except KeyError as e:
+            if relation_name not in self._meta.relations:
+                raise model.ModelError('{} is not a known relation'.format(relation_name)) from e
+            return []
 
     def relation_list(self, relation_id):
-        return self._relation_list_map[relation_id]
+        try:
+            return self._relation_list_map[relation_id]
+        except KeyError as e:
+            raise model.RelationNotFoundError from e
```

Comment:
Again, nice to see the model being tuned to reflect the real implementation. I suspect we'll have a lot of that in the near future.